src/app/modules/shared_modules/keycard_popup/internal/key_pair_migrate_failure_state.nim

```diff
@@ -11,9 +11,19 @@ proc delete*(self: KeyPairMigrateFailureState) =
 method executePrePrimaryStateCommand*(self: KeyPairMigrateFailureState, controller: Controller) =
   if self.flowType == FlowType.SetupNewKeycard or
     self.flowType == FlowType.MigrateFromKeycardToApp:
-      controller.terminateCurrentFlow(lastStepInTheCurrentFlow = true)
+      let profileMigrated = controller.getSelectedKeyPairIsProfile()
+      if not profileMigrated:
+        controller.terminateCurrentFlow(lastStepInTheCurrentFlow = true)
+        return
+      info "quit the app because of profile migration failure"
+      quit() # quit the app
```

Comment:
@IvanBelyakoff that's a handling of a profile keypair migration and in that case, because of re-encryption we have to do that, in case of a non-profile keypairs migration we don't do that.